Samples/resource-owner-password-credential/Angular2/src/newoidc/Controllers/ResourceController.cs

```diff
@@ -0,0 +1,36 @@
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using newoidc.Data;
+using newoidc.Models;
+using System.Threading.Tasks;
+
+namespace newoidc.Controllers
+{
+    [Authorize]
+    public class TestController : Controller
+    {
+        private ApplicationDbContext _context;
+       private UserManager<ApplicationUser> _userManager;
+
+        public TestController(ApplicationDbContext context, UserManager<ApplicationUser> userManager)
+        {
+            /*my cat did this
+            ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+            */
+            _context = context;
+            _userManager = userManager;
+        }
+
+        [Route("api/test"), HttpGet]
+        public async Task<IActionResult> Get()
+        {
+            var user = await _userManager.GetUserAsync(User);
```

Comment:
Since this controller has the [Authorize] attribute on it, the current user (the property User of the controller) is populated with all the claims that the token contains.

This is essentially a round-tripping a ClaimsIdentity, from Openiddict (as a ClaimsIdentity), to a JWT to re-hydrate a ClaimsIdentity.